plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConnector.java

```diff
@@ -0,0 +1,150 @@
+
+    @Inject
+    public FakerConnector(
+            FakerMetadata metadata,
+            FakerSplitManager splitManager,
+            FakerPageSourceProvider pageSourceProvider,
+            FakerPageSinkProvider pageSinkProvider)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.pageSourceProvider = requireNonNull(pageSourceProvider, "pageSourceProvider is null");
+        this.pageSinkProvider = requireNonNull(pageSinkProvider, "pageSinkProvider is null");
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly, boolean autoCommit)
+    {
+        return FakerTransactionHandle.INSTANCE;
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata(ConnectorSession session, ConnectorTransactionHandle transaction)
+    {
+        return metadata;
+    }
+
+    @Override
+    public ConnectorSplitManager getSplitManager()
+    {
+        return splitManager;
+    }
+
+    @Override
+    public ConnectorPageSourceProvider getPageSourceProvider()
+    {
+        return pageSourceProvider;
+    }
+
+    @Override
+    public ConnectorPageSinkProvider getPageSinkProvider()
+    {
+        return pageSinkProvider;
+    }
+
+    @Override
+    public Set<ConnectorCapabilities> getCapabilities()
+    {
+        return Set.of(NOT_NULL_COLUMN_CONSTRAINT);
+    }
+
+    @Override
+    public List<PropertyMetadata<?>> getSchemaProperties()
+    {
+        return ImmutableList.of(
+                doubleProperty(
+                        SchemaInfo.NULL_PROBABILITY_PROPERTY,
+                        "Default probability of null values in any column that allows them, in any table of this schema",
+                        null,
+                        nullProbability -> checkState(0 <= nullProbability && nullProbability <= 1, "null_probability value must be between 0 and 1, inclusive"),
+                        false),
+                longProperty(
+                        SchemaInfo.DEFAULT_LIMIT_PROPERTY,
+                        "Default limit of rows returned from any table in this schema, if not specified in the query",
+                        null,
+                        defaultLimit -> checkState(1 <= defaultLimit, "default_limit value must be equal or greater than 1"),
+                        false));
+    }
+
+    @Override
+    public List<PropertyMetadata<?>> getTableProperties()
+    {
+        return ImmutableList.of(
+                doubleProperty(
+                        TableInfo.NULL_PROBABILITY_PROPERTY,
+                        "Default probability of null values in any column in this table that allows them",
+                        null,
+                        nullProbability -> checkState(0 <= nullProbability && nullProbability <= 1, "null_probability value must be between 0 and 1, inclusive"),
```

Comment:
checkState is not the right exception here, we need to throw io.trino.spi.StandardErrorCode#INVALID_TABLE_PROPERTY